### ISSUE TITLE
Notify maintainers about problems with Bodhi update

### ISSUE THREAD

**Issue by lachmanfrantisek** [packit/packit-service#2404]
During discussions with the Gnome maintainers (https://github.com/packit/user/issues/18), a feature request appeared to be able to notify package maintainers about situations that require the maintainer's attention, e.g. negative karma, check failed,... but not about "normal" state like update created or positive karma added.

Alternative ideas:
* Fedora notifications
  * There is a way how to configure Fedora notifications via https://notifications.fedoraproject.org/
  * One can filter packages by maintainer/group.
  * There is a topic filtering available, but I am not sure if [Bodhi sends ](https://fedora-messaging.readthedocs.io/en/stable/user-guide/schemas.html#bodhi) negative karma comments to another topic.
* Since this might be a generic feature for any Bodhi user, we can think about the possibility of rather doing this in Bodhi itself.

UX:
* Ideally,  do this when Bodhi updates are configured.
* An [`issue_repository` config option](https://packit.dev/docs/configuration#issue_repository) might be used.

Implementation:
* This would require parsing new types of messages and a new handler.

**Comment by lbarcziova** [comment]
> Since this might be a generic feature for any Bodhi user, we can think about the possibility of rather doing this in Bodhi itself.

I would probably lean towards this option as it seems more natural to have it there to me.

**Comment by lachmanfrantisek** [comment]
Asking for opinion/help on Bodhi side: https://github.com/fedora-infra/bodhi/issues/5644

**Comment by nforro** [comment]
> * There is a topic filtering available, but I am not sure if [Bodhi sends ](https://fedora-messaging.readthedocs.io/en/stable/user-guide/schemas.html#bodhi) negative karma comments to another topic.

Karma updates are still comments, so `org.fedoraproject.prod.bodhi.update.comment` works, e.g. [negative karma](https://apps.fedoraproject.org/datagrepper/v2/id?id=76415e15-9276-4ad5-831f-674be7cc3ae4&is_raw=true&size=extra-large) for libarchive-3.7.2-5.fc40.

**Comment by ljavorsk** [comment]
> Karma updates are still comments, so org.fedoraproject.prod.bodhi.update.comment works, e.g. [negative karma](https://apps.fedoraproject.org/datagrepper/v2/id?id=76415e15-9276-4ad5-831f-674be7cc3ae4&is_raw=true&size=extra-large) for libarchive-3.7.2-5.fc40.

I didn't get any notification about negative karma on my package, which I didn't even think could happen, so if I hadn't been notified by @lachmanfrantisek I would not have noticed it for quite some time.


> An [issue_repository config option](https://packit.dev/docs/configuration#issue_repository) might be used.

Not sure about this option as some packages don't even configure this option. If you decide to use it, it should be heavily documented in your onboarding docs (so maintainers don't miss it). But I would rather see a regular Bodhi notification email, as I would expect from a regular update I create manually.

**Comment by kparal** [comment]
I included some examples of where this might cause problems in https://github.com/packit/packit/issues/2342 .

I'd also like to add that even some neutral comments might have content that the maintainer should see, perhaps a question or a concern, when the user is not sure whether it's a bug or not. One option could be if Bodhi implemented a "ping maintainer" checkbox when submitting feedback. Another option is to notify the maintainer (from packit side) also for neutral (and not just negative) comments. Or, for all comments (which is the current Bodhi behavior for manually created updates). Any solution would be an improvement over the current status.

**Comment by lbarcziova** [comment]
On a related note, in #2503 it was mentioned, that something similar would be useful also for downstream Koji builds - be notified for a failure. The preferred way there was an email.

**Comment by QuLogic** [comment]
I don't think you should wait to figure out "important" comments on Bodhi, but just get maintainers notified of Bodhi updates entirely first. Later you can wait for #5644.

**Comment by lbarcziova** [comment]
Outcome from our architecture discussion: we will also set up a meeting with someone from CPE to brainstorm how to improve the situation.

**Comment by lachmanfrantisek** [comment]
I've reached CPE team and was asked to create a ticket to start the discussion -- so, [here](https://pagure.io/fedora-infrastructure/issue/12234), it is. There are already a few interesting suggestions if you are interested in this. (Yet to decide what would be the best approach.)

**Comment by LecrisUT** [comment]
From the discussions at flock, we actually have a simple way to resolve this by just `@mention` the maintainers. Currently the suggestion is to use the account for who merged the PR as the person to be mentioned. There are services like KDE's discover that take the changelog from the bodhi update, so I don't think it should be in the first message but a second one.

@AdamWill also suggested that we could patch bodhi to support this [more natively](https://github.com/fedora-infra/bodhi/issues/6129), and I think it is very much worth it since there are other issues reported like being able to give karma to yourself. It would be more challenging though, so in the meantime I think we should go with the previous suggestion.

**Comment by lbarcziova** [comment]
Just for transparency, the topic was also discussed on the moved issue in https://forge.fedoraproject.org/infra/tickets/issues/12234.

**Comment by kparal** [comment]
> use the account for who merged the PR as the person to be mentioned

The author of the commit, or even the person who merges that commit, might not always be the maintainer. Can you ideally mention both the committer and the package maintainer (unless they are the same)? That should be the safest way to make sure relevant people are informed.

**Comment by LecrisUT** [comment]
> Can you ideally mention both the committer and the package maintainer (unless they are the same)?

Good point. That should also cover the `allowed_committers` workflow when they push directly also.
